Marlin/src/HAL/HAL_STM32F1/HAL_Servo_STM32F1.cpp

```diff
@@ -68,23 +68,25 @@ libServo::libServo() {
 
 bool libServo::attach(const int32_t pin, const int32_t minAngle, const int32_t maxAngle) {
   if (this->servoIndex >= MAX_SERVOS) return false;
+  if (!PWM_PIN(pin)) return false;
```

Comment:
This seems like something we should catch at compile time. Is that not possible?

Reply:
actually i changed that to do https://github.com/tpruvot/Marlin/commit/88ba25561836141611e423ec993677030bef9a34

Reply:
If the pin is constant at compile time, it can evaluate `PWM_PIN` at compile time. Note that HALs are free to define `PWM_PIN` as a function call, and then it would (perhaps) never be evaluated at compile time.